### DISCUSSION TITLE
Graph batching

### DISCUSSION THREAD

**Discussion by niko4299** [pyg-team/pytorch_geometric#3490] [Q&A] [answered]
Hey!

I just want to be sure if i do Batch.from_data_list([Data(x = x[i], edge_index = edge_index) for x in range(10)]) and then do graph convolution is graph convolution done for each separately and than output is concatenated ? I'm asking because i have 30 nodes and on output i get 300 values

Thank you for your help!

**Comment by wsad1** [comment, marked as answer]
`Batch.from_data_list([Data(x = x[i], edge_index = edge_index) for x in range(10)])` this outputs a single large graph(`Data` object) with 10 disconnected components, read more [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html#advanced-mini-batching). So your GNN gets one graph with 300 nodes(a graph with 10 disconnected components of 30 nodes each) and the convolution is perform on the whole graph with 300 nodes, and thats why your output has 300 values.

**Reply by niko4299** [reply]
Thank you for your answer!
